conseil-schema/src/main/scala/tech/cryptonomic/conseil/schema/GenSchema.scala

```diff
@@ -1,51 +1,147 @@
-    lazy val jdbcDriver = "org.postgresql.Driver"
-    lazy val slickProfile = "slick.jdbc.PostgresProfile"
-    lazy val `package` = "tech.cryptonomic.conseil.common.tezos"
-    lazy val dest = "/tmp/slick"
+    externallyProvided getOrElse classpathProvided
   }
 
-  //applies convention to uses CamelCase when reading config fields
-  implicit def hint[T] = ProductHint[T](ConfigFieldMapping(CamelCase, CamelCase))
-
-  val conseilDbConf = loadConfig[PostgresConfig](namespace = "conseil.db.properties")
-
-  conseilDbConf.foreach { cfg =>
-    println(s"Generating database Tables source file under ${cfg.dest}, in package ${cfg.`package`}")
-    slick.codegen.SourceCodeGenerator.main(
-      Array(
-        cfg.slickProfile,
-        cfg.jdbcDriver,
-        cfg.url,
-        cfg.dest,
-        cfg.`package`,
-        cfg.user,
-        cfg.password
-      )
+  io(uri = confUri, outputDir = None).unsafeRunSync()
+
+  /** Essentially follow along what the [[SourceCodeGenerator.run]] does,
+    * yet selecting only a number of tables, grouping the results in different
+    * source files based on the namespace.
+    *
+    * This keeps the different blockchains' db models separated, if we follow the
+    * namespace rule.
+    *
+    * @param uri will point to the configuration file, possibly in a subsection, using fragments
+    * @param outputDir we actually define this inside the configuration, but we could change that
+    * @return an IO effect to be executed via .unsafeRunSync() or similar calls
+    */
+  def io(
+      uri: URI,
+      outputDir: Option[String]
+  ): IO[Unit] = {
+    implicit val shift = IO.contextShift(ExecutionContext.global)
+
+    //we read configuration values
+    val dc = DatabaseConfig.forURI[PostgresProfile](uri)
+    val basePackage = dc.config.getString("codegen.package")
+    val out = outputDir orElse Option(dc.config.getString("codegen.outputDir")) getOrElse "."
+
+    /* Use cats.effect.Resource to get automatic-resource-management
+     * Wrapping the operations on resource content with a `resource.use` call
+     * will guarantee that those resources will be released after the execution.
+     */
+    val resources = for {
+      db <- Resource.fromAutoCloseable(IO(dc.db))
+      blocker <- Blocker[IO]
+    } yield (db, blocker)
+
+    resources.use {
+      case (db, blockingPool) =>
+        /* First we define the steps:
+         * - get the complete db model
+         * - extract schemas/namespaces from the model
+         * - filter the schemas we care about and write the sources for each
+         */
+        val getModel = IO.fromFuture(
+          IO.delay(
+            db.run(dc.profile.createModel(None)(ExecutionContext.global).withPinnedSession)
+          )
+        )
+
+        val getSchemas = (model: Model) =>
+          IO {
+            val schemas = model.tables.groupBy(_.name.schema).collect {
+              case (Some(schema), tables) if schema != "public" => (schema, tables)
```

Comment:
I think that "public" is not the only schema visible there. We still have `information_schema`, `pg_catalog` as a default schemas, or that totally different think and I'm confusing something?